Sources/Commands/SwiftTestTool.swift

```diff
@@ -578,6 +592,8 @@ struct UnitTest {
     }
 }
 
+extension UnitTest: Encodable {}
```

Comment:
if we go for an XUnit like format, we should probably create utility structs which will be the ones that are encoded. there will be a small cost of crating a second representation in memory, but I think it is worth it since it will give us a degree of freedom between the "real" model and the one we want to write to disk as JSON. its a technique I often use when build server APIs so I know it works well.

Reply:
fwiw, we can then also modify the XUnit generator to use these XUnit specific representations which will make that consistent across the two on-disk representations